packages/table/src/react/index.js

```diff
@@ -100,27 +110,44 @@ function ColumnHeader(props) {
   if (props.flex && !style.flex) style.flex = props.flex
 
   const handleClick = props.onClick
-    ? evt => props.onClick(getToggledSort(props), evt)
+    ? evt =>
+        props.onClick(
+          ...[evt, props.sort && getToggledSort(props)].filter(Boolean)
+        )
     : undefined
-
-  return (
-    <Tag
-      role="columnheader"
-      {...styles.columnHeader(themeName, props, { active })}
-      {...filterReactProps(props, { tagName: Tag })}
-      onClick={handleClick}
-      style={style}
-      aria-sort={
+  const children = props.onClick ? (
+    <button {...styles.columnHeaderButton()} onClick={handleClick}>
+      <div {...styles.columnHeaderButtonInner(props)}>
+        {props.children}
+        {props.sort && getSortIcon(props)}
+      </div>
+    </button>
+  ) : (
+    <>
+      {props.children}
+      {props.sort && getSortIcon(props)}
+    </>
+  )
+  const sort = {
```

Comment:
Looked at dequeuniversity and they weren't event showing this attribute if it wasn't sorted. I went halfway and basically am not applying it unless the user has set sort of some sort on the column.